src/sentry/search/snuba/backend.py

```diff
@@ -49,6 +50,19 @@ def unassigned_filter(unassigned, projects):
     return query
 
 
+def triaged_filter(triaged, projects):
+    query = Q(
+        id__in=GroupLink.objects.filter(
+            project_id__in=[p.id for p in projects],
+            linked_type=GroupLink.LinkedType.issue,
+            relationship=GroupLink.Relationship.references,
+        ).values_list("group_id", flat=True)
```

Comment:
Don't know how our search works but it _feels_ to me that using `.exists()` here would be more appropriate and efficient.

Reply:
I believe this is building onto a query that already does `SELECT ... FROM group`  and this is an additional filter for groups that is tacked on. `SELECT * FROM group [[WHERE id IN (...)]]`, where this function is basically providing an additional WHERE clause.

Using `exists()` to me in some capacity would imply we're doing a correlated subquery, .e.g`WHERE EXISTS (SELECT group_link WHERE group_id=group.id)`, which seems undesirable. I think theoretically doing a JOIN to group_link for link-existence and an OUTER JOIN for non-existence can work and allows the postgres planner to be more efficient, but in my experience that doesn't work well with query builders like this since it requires coordination between this filter and other filters (e.g. making sure the joins are aliased, or re-using pre-existing joins in the query).

FWIW, I'm pattern matching off of how assignees are built, so it should be roughly as efficient as that.